packages/data/src/onboarding/actions.js

```diff
@@ -217,6 +217,22 @@ export function* updateProfileItems( items ) {
 	}
 }
 
+function possiblyPruneTaskData( task, keys ) {
```

Comment:
Nice approach!  Should we leave a comment noting this can be removed after a certain version?

Reply:
Yeah I will add that